clang/test/3C/base_subdir/canwrite_constraints.c

```diff
@@ -14,7 +14,7 @@
 // because of https://github.com/correctcomputation/checkedc-clang/issues/503;
 // the rest of the test still works. TODO: Re-enable warning verification.
 //
-// RUN: cd %S && 3c -alltypes -addcr -output-dir=%t.checked/base_subdir -warn-all-root-cause %s --
+// RUN: cd %S && 3c -alltypes -addcr -output-dir=%t.checked/base_subdir -warn-all-root-cause %s -- -Xclang -verify
```

Comment:
This syntax suggests that only compile errors will be verified. Is the 3C verifier automatically activated in this case, or is it always on?

What would happen if only half the files had the -verify flag?

Reply:
Since the user of the diagnostics still needs to think in terms of translation units, even if we force all-or-none, I think this is a good solution as is. I was a little worried about our init script picking up a `-Xclang -verify` and upgrading it to 3c -verify, forcing an edit to that file. But that edit seems acceptable given our constraints. We just need to make sure it's in the 3c documentation that 3c will enhance -verify. I expect CCI will be the primary user of these options for a long time, so minimal documentation is fine.

Reply:
> I was a little worried about our init script picking up a `-Xclang -verify` and upgrading it to 3c -verify, forcing an edit to that file.

Do I understand your concern correctly: A user might have an existing file `foo.c` on which they run `clang -Xclang -verify foo.c` and it works, but then they try to run `3c foo.c -- -Xclang -verify` and it fails until they add `expected-*` directives for the additional diagnostics generated by 3C?  I can't think of what would motivate a user to do that.  Still, I agree that we should document the behavior of `-Xclang -verify` in 3C to help people understand how to write 3C tests, and that would also help a hypothetical user in the previous situation.

But where to document it?  IMO, `3c -help` would be too distracting to normal users.  `3c -help-hidden` (like the old `3c -verify` option) would seem reasonable to me, but I'm not aware of a way to add a message (as opposed to an option) that is displayed only by `-help-hidden` and not by `-help`.  Maybe [the Testing section of CONTRIBUTING.md](https://github.com/correctcomputation/checkedc-clang/blob/main/clang/docs/checkedc/3C/CONTRIBUTING.md#testing) then?